memCache/entryCache.go

```diff
@@ -387,8 +169,8 @@ func (e *Entry) refreshResponse(stableBackings []stableCache.Cache, client *http
 // refreshAndLog is a small wrapper around refreshResponse
 // for when a caller wants to run it in a goroutine and doesn't
 // want to handle the returned error itself
-func (e *Entry) refreshAndLog(stableBackings []stableCache.Cache, client *http.Client) {
-	err := e.refreshResponse(stableBackings, client)
+func (e *Entry) refreshAndLog(stableBackings []stableCache.Cache, client *http.Client, timeout time.Duration) {
```

Comment:
Instead of adding a timeout argument, what about starting to thread through `context` objects and set deadlines?